letsencrypt/storage.py

```diff
@@ -270,7 +270,10 @@ def current_target(self, kind):
             logger.debug("Expected symlink %s for %s does not exist.",
                          link, kind)
             return None
-        target = os.readlink(link)
+        if os.path.islink(link):
+            target = os.readlink(link)
```

Comment:
There's a slight race here, in theory the file could be deleted in between the check and the `readlink` call.

I could change this to do

```python
try:
    target = os.readlink(link)
except OSError:
    # Could add a check for more specific error here
    target = link
```